### PR TITLE
Make source_dir discoverable for non-src project

### LINKED ISSUE #2962: Make `source_dir` discoverable and document how to use this for non-standard project structure
## Description

Related to https://github.com/kedro-org/kedro/issues/2553
>Documentations: Document clearly what is pyproject.toml, How can you change pyproject.toml for [non-standard project without src](https://github.com/orgs/kedro-org/projects/2/views/19?pane=issue&itemId=20351285)
Current doc: https://docs.kedro.org/en/latest/kedro_project_setup/settings.html#project-metadata

We never document it clearly how to use `source_dir`. It's a misconception that `src` is a must for kedro-project.
https://github.com/kedro-org/kedro/blob/4563a4c609cf808e9087561af5619f143ba3fcff/kedro/framework/startup.py#L112-L123

## Context

Users misunderstood that `src` is a must for Kedro Project, it can be configured already but we never document it clearly.

## Possible Implementation

- Add documentations, i.e. How to work with Kedro if I have my own project structure already
- Add `source_dir=src` to `pyproject.toml` by default to make this discoverable.

### PR DESCRIPTION
**Pull request by noklam**
## Description

Fix #2962, surface the default settings in `pyproject.toml` to make is easier to discover.

The original default is hidden here:
https://github.com/kedro-org/kedro/blob/bdedb53f0cc3a766ae59f1222912d1cbf066df53/kedro/framework/startup.py#L104

## Development notes

- [ ] Changes in starters - https://github.com/kedro-org/kedro-starters/pull/202

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by AhdraMeraliQB** [comment]
Will this affect ProjectMetadata? Just thinking of the knock-on effects #3426 had when adding example-pipeline to `pyproject.toml`

**Comment by noklam** [comment]
@AhdraMeraliQB Good question! I don't think this will have an impact. Unlike the new starters/tool situation, `source_dir` is an old config and the default was always there:
https://github.com/kedro-org/kedro/blob/bdedb53f0cc3a766ae59f1222912d1cbf066df53/kedro/framework/startup.py#L104

The change is merely surface the config in a project

**Review by SajidAlamQB — APPROVED** [review]

File: docs/source/kedro_project_setup/settings.md [hunk 1]
```diff
@@ -31,6 +31,8 @@ Every Kedro project comes with a default pre-populated `pyproject.toml` file in
 package_name = "package_name"
 project_name = "project_name"
 kedro_init_version = "kedro_version"
+tools = ""
```
Comment (AhdraMeraliQB) [inline docs/source/kedro_project_setup/settings.md:34]: If we are including tools here we should probably include `example_pipeline` as well

**Review by AhdraMeraliQB — APPROVED** [review]
Looks good! Just one minor comment:

**Pull request merged**